sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -2015,4 +2015,121 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
       checkAnswer(table.filter($"p" === "p1\" and q=\"q1").select($"a"), Row(4))
     }
   }
+
+  test("TRIM function-BOTH") {
```

Comment:
For hive/execution/SQLQuerySuite, it will test the codeGen path (doGenCode). The source string is from column name.
 But I can remove those test cases not go through the codeGen path, both source string and trim string are string constant. 

`checkAnswer(sql("SELECT TRIM(BOTH '数$,.' FROM '数.数,数%') from trimBoth"), Row("%")) checkAnswer(sql("SELECT TRIM(BOTH '敠数' FROM '敠敠数敠敠') from trimBoth"), Row("")) checkAnswer(sql("SELECT TRIM(BOTH '搠abc' FROM 'a数b') from trimBoth"), Row("数"))`